src/mlpack/core/tree/vantage_point_tree/vantage_point_split_impl.hpp

```diff
@@ -0,0 +1,298 @@
+/**
+ * @file vantage_point_split_impl.hpp
+ * @author Mikhail Lozhnikov
+ *
+ * Implementation of class (VantagePointSplit) to split a binary space partition
+ * tree according to the median value of the distance to a certain vantage point.
+ */
+#ifndef MLPACK_CORE_TREE_BINARY_SPACE_TREE_VANTAGE_POINT_SPLIT_IMPL_HPP
+#define MLPACK_CORE_TREE_BINARY_SPACE_TREE_VANTAGE_POINT_SPLIT_IMPL_HPP
+
+#include "vantage_point_split.hpp"
+#include <mlpack/core/tree/bounds.hpp>
+
+namespace mlpack {
+namespace tree {
+
+template<typename BoundType, typename MatType>
+bool VantagePointSplit<BoundType, MatType>::
+SplitNode(const BoundType& bound, MatType& data, const size_t begin,
+    const size_t count, size_t& splitCol)
+{
+  typename BoundType::ElemType mu;
```

Comment:
Couldn't you just use `ElemType` here?  We should be guaranteed that `BoundType::ElemType == MatType::elem_type`.